src/util/network/teller/appTellerManager.ts

```diff
@@ -53,6 +57,27 @@ export type PeraTeller =
       type: "SIGN_TXN_CALLBACK_ERROR";
       error: string;
     }
+  | {
+      type: "SIGN_DATA";
+      signer: string;
+      chainId: AlgorandChainIDs;
+      data: PeraWalletArbitraryData[];
+    }
+  | {
+      type: "SIGN_DATA_CALLBACK";
+      signedData: {
+        txnId: string;
+        signedData: string;
+      }[];
+    }
```

Comment:
There is no transaction id in signed Data, right? what should be the return type here?
```suggestion
     {
      type: "SIGN_DATA_CALLBACK";
      signedData:   PeraWalletArbitraryData[]
    }
```